packages/mongodb-mql-model/src/docs/md/mql-query/mql-query.md

```diff
@@ -0,0 +1,145 @@
+so it can be tailored back to the original source code. A query MAY be unsupported by a specific target MongoDB Cluster.
+
+### Query validation and support
+
+A query MAY be valid for a target MongoDB Cluster if the MongoDB Cluster can consume the query once it
+is translated to a consumable dialect by the target Cluster. However, a query MAY be unsupported by the 
+cluster if it doesn't have the capabilities to fulfill the query request.
+
+For example, let's consider the following query, in pseudocode:
+
+```java
+collection.aggregate(AtlasSearch(text().eq("baby one more time")))
+```
+
+| Cluster 	                      | Is Valid 	 | Is Supported 	 |
+|--------------------------------|------------|----------------|
+| MongoDB Community 7.0          | ✅          | 	       🔴     |
+| MongoDB Enterprise 8.0	        | ✅	         | 	🔴            |
+| MongoDB Atlas 8.0 w/o Search	  | 	✅         | 	   🔴         |
+| MongoDB Atlas 8.0 with Search	 | 	✅         | 	   ✅          |
+
+For the purpose of this specification and project, we will only allow the `mongosh` dialect as a
+consumable dialect for a MongoDB Cluster.
+
+### Query equivalence
+
+We will consider two queries equivalent, independently of the query structure, if the following conditions
+apply:
+
+* They MUST be **valid** by the same set of target clusters.
+* They MUST be **supported** by the same set of target clusters.
+* They MUST return the same subset of results for the same input data set.
+* They MAY be sourced from the same dialect.
+* They MAY lead to equivalent **execution plans** for the same target cluster.
+
+We will consider two execution plans equivalent if the cluster query planner lead to the same list
+of operations.
+
+Let's consider a different use case. For the following two queries in the `Java Driver` dialect:
+
+```java
+collection.find(eq("bookName", myBookName))
+collection.aggregate(matches(eq("bookName", myBookName)))
+```
+
+We will test two different target clusters:
+
+* MongoDB Community 8.0, from a development environment, does not have an index on bookName for the target collection.
+* MongoDB Atlas 8.0, production environment, does have an index on bookName for the target collection.
+
+In the development environment, we copy the data from the production environment once every week. For this example,
+we will consider that the data sets are exactly the same on both clusters.
+
+| Cluster Environment           	 | Is Valid 	  | Is Supported 	 | Same Results 	 | Same Dialect 	 | Same Execution Plan 	 |
+|---------------------------------|-------------|----------------|----------------|----------------|-----------------------|
+| MongoDB Development         	   | ✅         	 | ✅        	     | ✅          	   | ✅        	     | 🔴                	   |
+| MongoDB Production 	            | ✅         	 | ✅           	  | ✅         	    | ✅        	     | 🔴                	   |
```

Comment:
What's an execution plan? Because, in the example here, they _would_ typically end up using the same queryPlan

Reply:
Well in this case, in Production we do have an index, but in Development we don't, so they lead to different execution plans (prod = IXSCAN, dev = COLLSCAN) but queries are still equivalent. That's what I was thinking. WDYT?

Reply:
Ok, so the point of this table isn't that they don't have the same execution plan between both queries (they do!), it's that they don't have the same execution plan between both environments?

Then I'd say it's a bit odd how this table works, normally when you have two rows there should either be a possibility of distinct values in each row or the cells for the rows should be merged to clarify that that can't happen

Reply:
Ah I get your point now. You are right. I'm thinking how else I can model this. I'll push a suggestion. Do you have a better idea in your mind?

Reply:
Just tried with a more complex table, but I think it's more clear.

Reply:
It is! :)